specs/_features/ePBS/p2p-interface.md

```diff
@@ -0,0 +1,119 @@
+
+
+New validation:
+
+- _[REJECT]_ The block's execution header timestamp is correct with respect to the slot -- i.e. `header.timestamp == compute_timestamp_at_slot(state, block.slot)`.
+- _[REJECT]_ The bid pubkey is a valid builder in state.
+- _[REJECT]_ The builder has sufficient balances to pay for the bid.
+- _[REJECT]_ The builder signature, `signed_bid.signature`, is valid with respect to the `bid.pubkey`.
+
+###### `execution_payload`
+
+This topic is used to propagate execution payload.
+
+The following validations MUST pass before forwarding the `execution_payload` on the network, assuming the alias `payload = execution_payload`:
+
+- _[IGNORE]_ The payload aligns with header that received in block.
+
+###### `execution_attestation`
+
+This topic is used to propagate signed execution attestation.
+
+The following validations MUST pass before forwarding the `execution_attestation` on the network, assuming the alias `data = execution_attestation.data`:
+
+- _[IGNORE]_ `data.slot` is within the last `ATTESTATION_PROPAGATION_SLOT_RANGE` slots (within a MAXIMUM_GOSSIP_CLOCK_DISPARITY allowance) -- i.e. `data.slot + ATTESTATION_PROPAGATION_SLOT_RANGE >= current_slot >= data.slot` (a client MAY queue future attestations for processing at the appropriate slot).
+- _[REJECT]_ The validator index is within the execution committee in `get_execution_committee(state, data.slot)`
+- _[IGNORE]_ The execution payload being voted for (`data.execution_hash`) has been seen (via both gossip and non-gossip sources) (a client MAY queue execution attestations for processing once payload is retrieved).
+- _[REJECT]_ The signature of `execution_attestation` is valid.
+
+###### `builder_bid`
+
+This topic is used to propagate signed builder bids.
+
+The following validations MUST pass before forwarding the `signed_builder_bid` on the network, assuming the alias `bid = signed_builder_bid.message`:
+
+- _[REJECT]_ The signed builder bid pubkey, `bid.pubkey`, exists in state.
+- _[IGNORE]_ The signed builder bid value, `bid.value`, is less than builder's balance in state.
+- _[IGNORE]_ The signed builder header timestamp is correct with respect to next slot -- i.e. `header.timestamp == compute_timestamp_at_slot(state, current_slot + 1)`.
+- _[IGNORE]_ The signed builder header parent block has matches one of the chain tip in the fork choice store. Builder may submit multiple bids with respect to forks.
+- _[REJECT]_ The builder signature, `signed_bid.signature`, is valid with respect to the `bid.pubkey`.
+
+#### Transitioning the gossip
+
+See gossip transition details found in the [Altair document](../altair/p2p-interface.md#transitioning-the-gossip) for
+details on how to handle transitioning gossip topics for this upgrade.
+
+### The Req/Resp domain
```

Comment:
Block by root and by range will propagate full blocks (`SignedBeaconBlock`) or blinded blocks `SignedBeaconBlockWithBid`

Reply:
This is a good question and I admit I haven't thought much on. 
I think by range requires no changes. It's applied to finalized blocks anyway
By root is tricky, we might want to add a `signed_blinded_block_by_root` method

Reply:
not sure why would signed by root be required for blinded  except for those on the tip, but should be pickedup from gossip, anything missed from gossip can't be synced next slot by the time the payload is revealed (assuming that this ePBS scheme has payload reveal before last interval of the slot)

Reply:
I think the notion of "full block" dissappears on ePBS. You request blocks and payloads separated, there's very little need to assemble a full block except for validation in which you get the two lists anyway as they perform separate state transitions on the beacon state.